framework/requirements.txt

```diff
@@ -56,6 +60,7 @@ mypy-extensions==0.4.3
 numpy==1.26.0
 openapi-schema-validator==0.6.2
 openapi-spec-validator==0.7.1
+opensearch-py==2.6.0
```

Comment:
I've read in the OpenSearch [documentation](https://github.com/opensearch-project/opensearch-py/blob/main/guides/async.md#asynchronous-io) that the dependency has to be `opensearch-py[async]` for asynchronous I/O communication. Not really sure if it is installed anyway, but we should double-check that

Reply:
Yes, but `opensearch-py[async]` isn't a package by itself. The `[async]` it's an optional set of [dependencies](https://github.com/opensearch-project/opensearch-py/blob/main/setup.py#L122).

When `opensearch-py[async]` it called these packages

<details><summary>pip install output</summary>
<p>

```console
# pip install opensearch-py[async]
Collecting opensearch-py[async]
  Using cached opensearch_py-2.6.0-py2.py3-none-any.whl.metadata (7.0 kB)
Collecting requests<3.0.0,>=2.4.0 (from opensearch-py[async])
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting six (from opensearch-py[async])
  Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Collecting python-dateutil (from opensearch-py[async])
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting certifi>=2022.12.07 (from opensearch-py[async])
  Using cached certifi-2024.7.4-py3-none-any.whl.metadata (2.2 kB)
Collecting Events (from opensearch-py[async])
  Using cached Events-0.5-py3-none-any.whl.metadata (3.9 kB)
Collecting urllib3!=2.2.0,<3,>=1.26.18 (from opensearch-py[async])
  Using cached urllib3-2.2.2-py3-none-any.whl.metadata (6.4 kB)
Collecting aiohttp<4,>=3.9.4 (from opensearch-py[async])
  Using cached aiohttp-3.9.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.5 kB)
Collecting aiosignal>=1.1.2 (from aiohttp<4,>=3.9.4->opensearch-py[async])
  Using cached aiosignal-1.3.1-py3-none-any.whl.metadata (4.0 kB)
Collecting attrs>=17.3.0 (from aiohttp<4,>=3.9.4->opensearch-py[async])
  Using cached attrs-23.2.0-py3-none-any.whl.metadata (9.5 kB)
Collecting frozenlist>=1.1.1 (from aiohttp<4,>=3.9.4->opensearch-py[async])
  Using cached frozenlist-1.4.1-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)
Collecting multidict<7.0,>=4.5 (from aiohttp<4,>=3.9.4->opensearch-py[async])
  Using cached multidict-6.0.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.2 kB)
Collecting yarl<2.0,>=1.0 (from aiohttp<4,>=3.9.4->opensearch-py[async])
  Using cached yarl-1.9.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (31 kB)
Collecting charset-normalizer<4,>=2 (from requests<3.0.0,>=2.4.0->opensearch-py[async])
  Using cached charset_normalizer-3.3.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (33 kB)
Collecting idna<4,>=2.5 (from requests<3.0.0,>=2.4.0->opensearch-py[async])
  Using cached idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Using cached aiohttp-3.9.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)
Using cached certifi-2024.7.4-py3-none-any.whl (162 kB)
Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Using cached urllib3-2.2.2-py3-none-any.whl (121 kB)
Using cached Events-0.5-py3-none-any.whl (6.8 kB)
Using cached opensearch_py-2.6.0-py2.py3-none-any.whl (311 kB)
Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Using cached attrs-23.2.0-py3-none-any.whl (60 kB)
Using cached charset_normalizer-3.3.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (140 kB)
Using cached frozenlist-1.4.1-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (272 kB)
Using cached idna-3.7-py3-none-any.whl (66 kB)
Using cached multidict-6.0.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (128 kB)
Using cached yarl-1.9.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (328 kB)
Installing collected packages: Events, urllib3, six, multidict, idna, frozenlist, charset-normalizer, certifi, attrs, yarl, requests, python-dateutil, aiosignal, opensearch-py, aiohttp
Successfully installed Events-0.5 aiohttp-3.9.5 aiosignal-1.3.1 attrs-23.2.0 certifi-2024.7.4 charset-normalizer-3.3.2 frozenlist-1.4.1 idna-3.7 multidict-6.0.5 opensearch-py-2.6.0 python-dateutil-2.9.0.post0 requests-2.32.3 six-1.16.0 urllib3-2.2.2 yarl-1.9.4
```

</p>
</details>  

```console
# pip freeze
aiohttp==3.9.5
aiosignal==1.3.1
attrs==23.2.0
certifi==2024.7.4
charset-normalizer==3.3.2
Events==0.5
frozenlist==1.4.1
idna==3.7
multidict==6.0.5
opensearch-py==2.6.0
python-dateutil==2.9.0.post0
requests==2.32.3
six==1.16.0
urllib3==2.2.2
yarl==1.9.4
```